### DISCUSSION TITLE
Grant photo permission in iOS using xcrun simctl privacy <device> grant photos app.id

### DISCUSSION THREAD

**Discussion by antranapp** [fastlane/fastlane#18900] [Q&A]
I'm trying to automate my workflow to generate appstore screenshots. 

My app needs photo permission to access photos and videos, so it will needs photo permission.

I can grant photo permission from command line using `xcrun simctl privacy <device> grant photos app.id`, but I don't know how to hook this into fastlane workflow.

As far as I can see, fastlane can run `addphotos` and `addvideos` after device is booted, how can I run the `xcrun simctl` command before running `snapshot` without messing around with Ruby code? I have no idea about Ruby :(

**Comment by minuscorp** [comment]
I'd just `sh` your needed command to grant access and that should be it. No messing around 😁

**Reply by antranapp** [reply]
how can I get the id of the booted simulator in Fastfile? To run that shell command, I need the simulator id.

Also, how can I know when the simulator is done with booting to run this command?

Have you tried this by yourself? Can you post your solution?

**Reply by hicnar** [reply]
Reed this article https://itnext.io/you-dont-need-gui-or-how-to-control-ios-simulator-from-command-line-bf5cfa60aed2
In it the author shows how to build a shell script with some basic functionality you will need to control your iOS simulator from  command line. It includes a function for finding the UUID of a simulator based on the simulator's name.

And here's how I wait for my sims to be fully booted (#!/bin/bash) that function is not in the script from the above article.

function wait_till_booted()
{
    SIM_UUID="${1}"
    RESULT=$(xcrun simctl spawn "${SIM_UUID}" launchctl print system | grep com.apple.springboard.services)
    counter=$((0))
    while [[ "$RESULT" = "" || ! -f "$HOME/Library/Developer/CoreSimulator/Devices/$SIM_UUID/data/Library/Preferences/.GlobalPreferences.plist" ]]; do
        sleep 2
        ((counter++))
        RESULT=$(xcrun simctl spawn "${SIM_UUID}" launchctl print system | grep com.apple.springboard.services)
        echo -e "\033[34mWaiting on boot event for device ${SIM_UUID}, RESULT = ${RESULT}\033[0m"
        if [ "$counter" -gt 90 ]; then
            echo -e "\033[31mBooting of the device ${SIM_UUID} failed on timeout\033[0m"
            exit 1
        fi
    done
    echo "Device ${SIM_UUID} booted successfully"
}
